tee-worker/scripts/litentry/release/dev_deploy.sh

```diff
@@ -136,6 +125,31 @@ function main {
     esac
   done
 
+  # TODO: check flags conflict, e.g.
+  # - having `--discard` together with `upgrade-worker` doesn't make sense
+  # - `upgrade-worker` should ignore the `--only-worker` flag
+  if [[ "$DISCARD" == true && "$ACTION" == "upgrade-worker" ]]; then
+      echo "Option \"--discard\" and subcommand \"upgrade-worker\" should not be used together"
+      exit 1
+  fi
+
+
+  # 1/ check if $USER has sudo
+  if sudo -l -U $USER 2>/dev/null | grep -q 'may run the following'; then
+    source "$SGX_SDK/environment"
+  else
+    echo "$USER doesn't have sudo permission"
+    exit 1
+  fi
```

Comment:
That won't work though

root has uid 0, but non-zero uid doesn't mean it can't sudo. In fact, the code will always into `else` branch unless executed with root.